diagnostics/include/performance_handler.h

```diff
@@ -353,9 +353,14 @@ struct HistogramData {
 
 class PerformanceHandler {
  public:
+  /*
+  If a component is already registered: print a warning and do nothing. This is the API definition, there is nothing
+  ill-defined here. For example std::vector::clear(), std::unique_ptr::reset() or std::set::insert() operates in the
+  same way. Same idea for the unregistered part.
+  */
```

Comment:
Please add to the PR description what you fixed in PerformanceHandler. Ideally, it should be a different PR.

Reply:
@yuliasherman , I agree it would have been better to have on another PR but I found this issue as part of my PR, some test crashed I beleive.
I've added a description on the above merge commit description.
Thanks.